R-package/R/create-release-files.R

```diff
@@ -0,0 +1,411 @@
+  }
+  
+  return(list(dictionary = dictionary,
+              dictionary_version = summary_version))
+}
+
+#' Validate files sheet
+#'
+#' Validate files sheet and its internal content for proper file creation.
+#'
+#' @param dictionary_name string containing the file dictionary name.
+#' @param version string containing the dictionary version.
+#' @param dictionary_path string containing path to the dictionary directory.
+#'
+#' @return 2 lists containing csvs to export and another list containing excels to export.
+validate_files_sheet <-
+  function(dictionary,
+           version) {
+    files_sheet <-
+      openxlsx::readWorkbook(dictionary, odm_dictionary$files_sheet_name)
+    sets_sheet <-
+      openxlsx::readWorkbook(dictionary, odm_dictionary$sets_sheet_name)
+    parts_sheet <-
+      openxlsx::readWorkbook(dictionary, odm_dictionary$parts_sheet_name)
+    
+    # Remove any rows with not supported file_type.
+    files_sheet_formatted <-
+      files_sheet[files_sheet[[files$file_type$name]] %in% files$file_type$categories, unlist(files_sheet_column_names)]
+    # insert version
+    files_sheet_formatted$name <-
+      gsub('\\{version\\}', version, files_sheet_formatted$name)
+    files_sheet_formatted[[files$add_headers$name]] <-
+      gsub('\\{version\\}', version, files_sheet_formatted[[files$add_headers$name]])
+    files_sheet_formatted[[files$osf_locations$name]] <-
+      gsub('\\{version\\}', version, files_sheet_formatted[[files$osf_locations$name]])
+    files_to_extract <- list()
+    errors <- FALSE
+    for (row_index in seq_len(nrow(files_sheet_formatted))) {
+      working_row <- files_sheet_formatted[row_index,]
+      
+      # File extraction info
+      fileID <- working_row[[files$file_id$name]]
+      file_name <- working_row[[files$file_name$name]]
+      file_type <- working_row[[files$file_type$name]]
+      partID <- working_row[[files$part_ID$name]]
+      add_headers <- working_row[[files$add_headers$name]]
+      destination <- working_row[[files$destinations$name]]
+      osf_location <- working_row[[files$osf_locations$name]]
+      github_location <- working_row[[files$github_location$name]]
+      
+      validated_file <- FALSE
+      
+      # Validate destination
+      if (!(destination %in% files$destinations$categories)) {
+        logger::log_warn(paste0("File ID: ",
+                                fileID,
+                                " has an invalid destination set"))
+        errors <- TRUE
+      }
+      
+      set_info <- sets_sheet[sets_sheet$setID == partID, ]
+      # Determine if the partID supplied is set or part.
+      if (file_type == files$file_type$categories$excel) {
+        # If part exists in sets its a set therefore all parts belonging to the set are used as partID for sheet creation.
+        if (nrow(set_info) >= 1) {
+          set_name <- partID
+          set_parts <- set_info[["partID"]]
+          partID <- set_parts
```

Comment:
Why are you re-assigning this variable here?

Reply:
partID should be renamed since it is a confusing name in this context.